pkg/curatedpackages/bundle.go

```diff
@@ -76,43 +86,45 @@ func (b *BundleReader) getActiveBundleFromCluster(ctx context.Context) (*package
 	return bundle, nil
 }
 
-func (b *BundleReader) getPackageBundle(ctx context.Context, activeBundle string) (*packagesv1.PackageBundle, error) {
-	params := []string{"get", "packageBundle", "-o", "json", "--kubeconfig", b.kubeConfig, "--namespace", constants.EksaPackagesName, activeBundle}
-	stdOut, err := b.kubectl.ExecuteCommand(ctx, params...)
+func (b *BundleReader) getPackageBundle(ctx context.Context, activeBundle string) (
+	*packagesv1.PackageBundle, error,
+) {
+	pb := &packagesv1.PackageBundle{}
+	err := b.restClient.Get().Namespace(constants.EksaPackagesName).
+		Resource("packagebundles").Name(activeBundle).Do(ctx).Into(pb)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("getting package bundle: %w", err)
 	}
-	obj := &packagesv1.PackageBundle{}
-	if err := json.Unmarshal(stdOut.Bytes(), obj); err != nil {
-		return nil, fmt.Errorf("unmarshaling package bundle: %w", err)
-	}
-	return obj, nil
+
+	return pb, nil
 }
 
-func (b *BundleReader) GetActiveController(ctx context.Context) (*packagesv1.PackageBundleController, error) {
-	params := []string{"get", "packageBundleController", "-o", "json", "--kubeconfig", b.kubeConfig, "--namespace", constants.EksaPackagesName, packagesv1.PackageBundleControllerName}
-	stdOut, err := b.kubectl.ExecuteCommand(ctx, params...)
+func (b *BundleReader) GetActiveController(ctx context.Context) (
+	*packagesv1.PackageBundleController, error,
+) {
+	pbc := &packagesv1.PackageBundleController{}
+	err := b.restClient.Get().Namespace(constants.EksaPackagesName).
+		Resource("packagebundlecontrollers").
+		Name(packagesv1.PackageBundleControllerName).Do(ctx).Into(pbc)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("getting package bundle controller: %w", err)
 	}
-	obj := &packagesv1.PackageBundleController{}
-	if err := json.Unmarshal(stdOut.Bytes(), obj); err != nil {
-		return nil, fmt.Errorf("unmarshaling active package bundle controller: %w", err)
-	}
-	return obj, nil
+
+	return pbc, nil
 }
 
 func (b *BundleReader) UpgradeBundle(ctx context.Context, controller *packagesv1.PackageBundleController, newBundle string) error {
 	controller.Spec.ActiveBundle = newBundle
-	controllerYaml, err := yaml.Marshal(controller)
-	if err != nil {
-		return err
-	}
-	params := []string{"apply", "-f", "-", "--kubeconfig", b.kubeConfig}
-	_, err = b.kubectl.CreateFromYaml(ctx, controllerYaml, params...)
+	err := b.restClient.Put().
```

Comment:
IMO this is ok for this usecase. But worth noting we are changing the behavior here from moving from apply (3 way merge, etc.) to PUT.

Reply:
Interesting. I guess I'm not 100% up to date on the semantics of the POST/PUT/PATCH vs kubectl commands. Do you happen to have a good link to docs on that so I can brush up? I'd be curious how the command line is handling 3-way merges and general race issues.

I guess I should read the kubectl code.